### PR TITLE
T5559: Add static neighbor-proxy feature

### PR DESCRIPTION
**Pull request by sever-sever**
## Change Summary

Ability to set neighbor proxy ARP/NDP for specific hosts
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T5559

## Component(s) name

proxy-arp, proxy-ndp
## How to test
VyOS configuration:
```
set protocols static neighbor-proxy arp 192.0.2.1 interface 'eth1'
set protocols static neighbor-proxy arp 192.0.2.1 interface 'eth0'
set protocols static neighbor-proxy arp 192.0.2.2 interface 'eth0'
set protocols static neighbor-proxy arp 192.0.2.3 interface 'eth0'
set protocols static neighbor-proxy nd 2001:db8::1 interface 'eth1'

```
Check:
```
vyos@r4# sudo ip neighbor show proxy
192.0.2.2 dev eth0 proxy 
192.0.2.3 dev eth0 proxy 
192.0.2.1 dev eth0 proxy 
192.0.2.1 dev eth1 proxy 
2001:db8::1 dev eth1 proxy 

```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by c-po** [comment]
What about:

* `set protocols static arp 192.0.2.1 interface eth0` as an extension to the current static ARP <-> MAC CLI nodes we have?
* `set protocols static ndp 2001:db8::1 interface eth1`
?

**Comment by c-po** [comment]
What about to summarize all under:

```
set protocols static neighbor-proxy arp 192.0.2.1 interface eth0
set protocols static neighbor-proxy arp 192.0.2.1 interface eth1
set protocols static neighbor-proxy nd 2001:db8::1 interface eth1
```

Then we will have a common root node for the feature for both address families. Does this also work for VRFs?

**Comment by sever-sever** [comment]
> What about to summarize all under:
> 
> ```
> set protocols static neighbor-proxy arp 192.0.2.1 interface eth0
> set protocols static neighbor-proxy arp 192.0.2.1 interface eth1
> set protocols static neighbor-proxy nd 2001:db8::1 interface eth1
> ```
> 
> Then we will have a common root node for the feature for both address families. Does this also work for VRFs?

@c-po  Done!
```
set protocols static neighbor-proxy arp 192.0.2.1 interface 'eth1'
set protocols static neighbor-proxy arp 192.0.2.1 interface 'eth0'
set protocols static neighbor-proxy arp 192.0.2.2 interface 'eth0'
set protocols static neighbor-proxy arp 192.0.2.3 interface 'eth0'
set protocols static neighbor-proxy nd 2001:db8::1 interface 'eth1'

```
check
```
vyos@r4#  sudo ip neighbor show proxy
192.0.2.2 dev eth0 proxy 
192.0.2.3 dev eth0 proxy 
192.0.2.1 dev eth0 proxy 
192.0.2.1 dev eth1 proxy 
2001:db8::1 dev eth1 proxy
```
Not sure about VRF, it was a request from the task

File: interface-definitions/protocols_static_neighbor-proxy.xml.in [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                  <valueHelp>
+                    <format>ipv6</format>
+                    <description>IPv6 destination address</description>
+                  </valueHelp>
+                  <constraint>
+                    <validator name="ipv6-address"/>
+                  </constraint>
+                </properties>
+                <children>
+                  <leafNode name="interface">
```
Comment (c-po) [inline interface-definitions/protocols_static_neighbor-proxy.xml.in]: We could use an include for the interface here

File: src/conf_mode/protocols_static_neighbor-proxy.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    return config
+
+
+def verify(config):
+
+    if 'arp' in config:
+        for neighbor, neighbor_conf in config['arp'].items():
+            if 'interface' not in neighbor_conf:
+                raise ConfigError(f'neighbor-proxy arp {neighbor} interface required but not set.')
```
Comment (c-po) [inline src/conf_mode/protocols_static_neighbor-proxy.py]: Could you please reword the error message: „ARP neighbor-proxy for „IP“ requires an interface to be set!“
For both IPv4 and IPv6

**Review by c-po — APPROVED** [review]

File: src/conf_mode/protocols_static_neighbor-proxy.py (hunk 2 above)
Comment (sever-sever) [reply src/conf_mode/protocols_static_neighbor-proxy.py]: The reason that I use it
```
neighbor-proxy arp {neighbor} interface
```
It is the path of configuration that is not valid.  It is clear where in the configuration you should add `interface`.

But sure, I can

Comment (sever-sever) [reply src/conf_mode/protocols_static_neighbor-proxy.py]: Done!

File: interface-definitions/protocols_static_neighbor-proxy.xml.in (hunk 1 above)
Comment (sever-sever) [reply interface-definitions/protocols_static_neighbor-proxy.xml.in]: Done!

**Comment by sever-sever** [comment]
@MergifyIo backport sagitta

**Pull request merged**